src/main/java/org/folio/rest/impl/TenantRefAPI.java

```diff
@@ -126,6 +120,19 @@ String servicePointUserFilter(String s) {
               .add("users", "service-points-users");
           }
           return tl.perform(attributes, headers, vertxContext, superRecordsLoaded);
-        });
+        }).onSuccess(handler -> {
+
+          if (ShelvingOrderUpdate.isModuleUpgradeRequest(attributes)) {
+            // Updates items with shelving order property. Only for version before the 19.5.0.
+            long startTime = System.currentTimeMillis();
+            new ShelvingOrderUpdate(5000)
+                .updateItems(attributes, headers, vertxContext)
+                .onSuccess(h -> log.info("Shelving order updates completed in {} seconds",
+                    (System.currentTimeMillis() - startTime) / 1000))
+                .onFailure(
+                    h -> log.error("Error occurs during shelving order update: {}", h.getMessage()));
+          }
+      });
   }
+
```

Comment:
Do you mean that using compose with future will provide possibility to return failure status of load data process by set future's result in case if items update somehow fails?

Reply:
yes.